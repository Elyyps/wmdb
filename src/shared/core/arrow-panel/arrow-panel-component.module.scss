.arrow-panel {
  display: block;
  position: relative;
  background-color: $global-background;
  border-left: 4px solid $brand-primary;
  padding: 13px 40px 14px 13px;
  margin-bottom: 1px;
  text-align: left;
  color: $global-color;
  transition: background-color 0.2s;
  @media (min-width: $breakpoint-small) {
    padding: 20px 40px 20px 20px;
  }

  &__title {
    font-size: 14px;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    margin: 0;
  }

  &__arrow {
    position: absolute;
    top: 50%;
    right: 16px;
    transform: translateY(-50%) rotate(-90deg);
    color: $brand-primary;

    path {
      stroke: currentColor;
      fill: none !important;
    }
  }

  &:hover {
    text-decoration: none;
    color: $global-inverse-color;
    background-color: $brand-primary;
    .arrow-panel__arrow {
      color: $global-inverse-color;
      path {
        fill: none;
      }
    }
  }
}
