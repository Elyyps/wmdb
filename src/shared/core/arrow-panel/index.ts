export * from "./arrow-panel.component";
