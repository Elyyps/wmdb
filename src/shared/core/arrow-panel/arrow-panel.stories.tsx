import React from "react";
import { storiesOf } from "@storybook/react";

import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { ArrowPanel } from "./arrow-panel.component";

storiesOf("Arrow panel", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <ArrowPanel title="Title" url="/" />);
