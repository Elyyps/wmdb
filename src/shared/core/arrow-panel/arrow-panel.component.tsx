import * as React from "react";
import { Link } from "react-router-dom";
import styles from "./arrow-panel-component.module.scss";
import { IconComponent } from "@app/prep/modules-prep/core";
import Arrow from "@assets/icons/chevron-down.svg";

interface IArrowPanelProps {
  link: string;
  title?: string;
}

const ArrowPanel = ({ link, title }: IArrowPanelProps) => (
  <Link to={link} className={styles["arrow-panel"]}>
    <div className={styles["arrow-panel__title"]}>{title}</div>
    <div className={styles["arrow-panel__arrow"]}>
      <IconComponent icon={Arrow} size={"14px"} stroke />
    </div>
  </Link>
);

export { ArrowPanel };
