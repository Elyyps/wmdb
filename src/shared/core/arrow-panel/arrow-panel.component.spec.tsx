import React from "react";
import { shallow } from "enzyme";
import { ArrowPanel } from "./arrow-panel.component";

describe("[ArrowPanel]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<ArrowPanel url="/" title="Title" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
