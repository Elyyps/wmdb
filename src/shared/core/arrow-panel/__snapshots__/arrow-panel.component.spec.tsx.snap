// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ArrowPanel] should render component without crashing 1`] = `
<Link
  className="arrow-panel"
  to="/"
>
  <div
    className="arrow-panel__title"
  >
    Title
  </div>
  <div
    className="arrow-panel__arrow"
  >
    <IconComponent
      fillColor="white"
      icon="chevron-down.svg"
      size="14px"
    />
  </div>
</Link>
`;
